src/app/services/sources.ts

```diff
@@ -56,7 +58,7 @@ export const sources = (api: KumaApi) => {
         tag: [`kuma.io/service:${service}`],
       })
 
-      return items.length > 0 ? items[0] : null
+      return items.length > 0 ? ExternalService.fromObject(items[0]) : null
```

Comment:
How much of a knock on effect would it be to remove this `null` here? (preferably replace it with a `[]`)

Reply:
First of all, this is not returning multiple items. It can only ever be one or none, so `[]` doesn’t work.

This is an interesting case: in `src/app/services/components/ExternalServiceDetails.vue:15` we check against `null` to show a specific empty state. However, we can’t easily change this to `undefined` because we’ve chosen for `undefined` to mean a `DataSource` is loading.

So this will need some more thought.